modules/core/misc/python/package/mat_wrapper/__init__.py

```diff
@@ -4,15 +4,16 @@
 import cv2 as cv
 from typing import TYPE_CHECKING, Any
 
-# Same as cv2.typing.NumPyArrayGeneric, but avoids circular dependencies
+# Same as cv2.typing.NumPyArrayNumeric, but avoids circular dependencies
 if TYPE_CHECKING:
-    _NumPyArrayGeneric = np.ndarray[Any, np.dtype[np.generic]]
+    _NumPyArrayNumeric = np.ndarray[Any, np.dtype[np.integer[Any] | np.floating[Any]]]
```

Comment:
Should this also include `np.unsignedinteger[Any]` ?

Reply:
NumPy integer is a base class for all integer types. In my understanding for both signed and unsigned ones.

Documentation: https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.integer